include/wil/registry.h

```diff
@@ -17,16 +17,2704 @@
+         * \return A wil::shared_cotaskmem_string created from the string value read from the registry
+         *         note, the returned string will have already passed through ExpandEnvironmentStringsW
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures, including value not found
+         */
+        template <>
+        inline ::wil::shared_cotaskmem_string get_value_expanded_string<::wil::shared_cotaskmem_string>(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name)
+        {
+            ::wil::shared_cotaskmem_string value;
+            const reg_view_details::reg_view regview{ key };
+            regview.get_value(subkey, value_name, value, REG_EXPAND_SZ);
+            return value;
+        }
+
+        /**
+         * \brief Reads a REG_EXPAND_SZ value, returning a wil::shared_cotaskmem_string
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param value_name A string specifying the name of the registry value to read from
+         *        can be nullptr to read from the unnamed default registry value
+         * \return A wil::shared_cotaskmem_string created from the string value read from the registry
+         *         note, the returned string will have already passed through ExpandEnvironmentStringsW
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures, including value not found
+         */
+        template <>
+        inline ::wil::shared_cotaskmem_string get_value_expanded_string<::wil::shared_cotaskmem_string>(HKEY key, _In_opt_ PCWSTR value_name)
+        {
+            return wil::reg::get_value_expanded_string<::wil::shared_cotaskmem_string>(key, nullptr, value_name);
+        }
+#endif // #if defined(__WIL_OBJBASE_H_STL)
+#endif // defined(__WIL_OBJBASE_H_)
+
+#if defined(_VECTOR_)
+        /**
+         * \brief Reads a registry value of the specified type, returning a std::vector<BYTE>
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param subkey A string to append to the HKEY to attempt to read from
+         *        can be nullptr if not needed
+         * \param value_name A string specifying the name of the registry value to read from
+         *        can be nullptr to read from the unnamed default registry value
+         * \param type The registry type for the specified registry value to read from - see RegGetValueW
+         * \return A std::vector<BYTE> containing the bytes of the specified registry value
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures, including value not found
+         */
+        inline ::std::vector<BYTE> get_value_byte_vector(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, DWORD type)

```

Comment:
should we use `std::byte` instead of `BYTE`?

Reply:
that would require more stl header requirements and more special handling for compiler support for __cpp_lib_byte :(

Reply:
@ChrisGuzak, any thoughts?

Reply:
BTW, we adapted the `set_` methods to accept any byte-like type. This getter still returns `std::vector<BYTE>`, though, which I think is acceptable. `BYTE` is pretty close to the "optimal" type here, too.

I'd still appreciate @ChrisGuzak your thoughts here.